src/handlers/imageGenerationsHandler.ts

```diff
@@ -0,0 +1,43 @@
+import { constructConfigFromRequestHeaders, tryTargetsRecursively } from "./handlerUtils";
+import { Context } from "hono";
+
+/**
+ * Handles the '/images/generations' API request by selecting the appropriate provider(s) and making the request to them.
+ *
+ * @param {Context} c - The Cloudflare Worker context.
+ * @returns {Promise<Response>} - The response from the provider.
+ * @throws Will throw an error if no provider options can be determined or if the request to the provider(s) fails.
+ * @throws Will throw an 500 error if the handler fails due to some reasons
+ */
+export async function imageGenerationsHandler(c: Context): Promise<Response> {
+    try {
+        let request = await c.req.json();
+        let requestHeaders = Object.fromEntries(c.req.raw.headers);
+        const camelCaseConfig = constructConfigFromRequestHeaders(requestHeaders)
+
+        const tryTargetsResponse = await tryTargetsRecursively(
+            c,
+            camelCaseConfig,
+            request,
+            requestHeaders,
+            "imageGenerate",
+            "POST",
+            "config"
+        );
+
+        return tryTargetsResponse;
+    } catch (err: any) {
+        console.log("completion error", err.message);
```

Comment:
Nitpick: Please change the error log to `console.log("imageGenerate error", err.message);`

Reply:
Pushed a fix